packages/console/src/components/Tooltip/index.tsx

```diff
@@ -1,57 +1,74 @@
-import { Nullable } from '@silverhand/essentials';
 import classNames from 'classnames';
-import React, { ReactNode, RefObject, useEffect, useState } from 'react';
+import React, { ReactNode, RefObject, useEffect, useLayoutEffect, useRef, useState } from 'react';
 import { createPortal } from 'react-dom';
 
+import usePosition from '@/hooks/use-position';
+
 import * as styles from './index.module.scss';
 
-type Props<T> = {
+type Props = {
   content: ReactNode;
-  domRef: RefObject<Nullable<T>>;
+  anchorRef: RefObject<Element>;
   className?: string;
-  behavior?: 'visibleOnHover' | 'visibleByDefault';
-};
-
-type Position = {
-  top: number;
-  left: number;
+  isKeepOpen?: boolean;
+  isSuccessful?: boolean;
```

Comment:
This `isSuccessful` prop seems unnecessary. All it does is adding some additional styles to the tooltip element, which can be done through `className` as well

Reply:
* We cannot apply custom classNames to the tooltip, because this tooltip is appended to the document and it is separate from the component where we use it.
* the old solution is no longer valid also because we have more than one arrow direction now and the `pseudo-element` of the CSS is hard to take effect.

Reply:
I don't think so... Gimme some time, I'll
 try

Reply:
I'm working on get-started dropdown yesterday and the dropdown is also appended directly to the document. Yet still I'm able to apply custom styles through `className`

Reply:
We also need to know the placement of the widget to write the style for the arrow, but the placement will change, we cannot write static styles for dynamic contents unless we export the tooltip state.

Reply:
If I understand correctly, the `successful` styles has nothing to do with positions, right? It just adds a green color effect to the background

Reply:
No, the arrow color is not `background`.

Reply:
I know, it's using border color, but still... pretty much the same

Reply:
Update 👍